OpenRA.Mods.Common/MapFormats/DefaultMap.cs

```diff
@@ -0,0 +1,1450 @@
+#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using OpenRA.FileFormats;
+using OpenRA.FileSystem;
+using OpenRA.Graphics;
+using OpenRA.Primitives;
+using OpenRA.Support;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.MapFormats
+{
+	public class DefaultMapLoader : IMapLoader
+	{
+		public readonly string Type;
+		public readonly IReadOnlyDictionary<string, MiniYaml> Metadata;
+
+		public Map Load(ModData modData, IReadOnlyPackage package)
+		{
+			return new DefaultMap(modData, package);
+		}
+
+		public Map Create(ModData modData, ITerrainInfo terrainInfo, int width, int height)
+		{
+			return new DefaultMap(modData, terrainInfo, width, height);
+		}
+
+		public string ComputeUID(ModData modData, IReadOnlyPackage package)
+		{
+			return DefaultMap.ComputeUID(package);
+		}
+
+		public void UpdatePreview(ModData modData, MapPreview mp, IReadOnlyPackage p, IReadOnlyPackage parent, MapClassification classification, string[] mapCompatibility, MapGridType gridType)
```

Comment:
This is a terrible name. This doesn't update, but rather sets Map values from MapPreview